yaksh/evaluator_tests/test_python_evaluation.py

```diff
@@ -7,47 +7,105 @@ class PythonEvaluationTestCases(unittest.TestCase):
+                                "expected_answer": "5",
+                                "test_id": u'null',
+                                "pos_args": ["3", "2"],
+                                "kw_args": {}
                                 }]
-        self.timeout_msg = ("Code took more than {0} seconds to run. "
-                            "You probably have an infinite loop in your code.").format(SERVER_TIMEOUT)
 
     def test_correct_answer(self):
-        user_answer = "def add(a, b):\n\treturn a + b"""
-        get_class = PythonCodeEvaluator(self.test_case_data, self.test, self.language, user_answer, ref_code_path=None, in_dir=None)
-        result = get_class.evaluate()
+        user_answer = "def add(a, b):\n\treturn a + b"
+        evaluator = PythonCodeEvaluator(self.test_case_data, self.test, self.language, user_answer, ref_code_path=None, in_dir=None)
+        result = evaluator.evaluate()
         self.assertTrue(result.get("success"))
         self.assertEqual(result.get("error"), "Correct answer")
 
     def test_incorrect_answer(self):
         user_answer = "def add(a, b):\n\treturn a - b"
-        test_case_data = [{"func_name": "add", 
-                                 "expected_answer": "5", 
-                                 "test_id": u'null', 
-                                 "pos_args": ["3", "2"], 
-                                 "kw_args": {}
-                                }]
-        get_class = PythonCodeEvaluator(self.test_case_data, self.test, self.language, user_answer, ref_code_path=None, in_dir=None)
-        result = get_class.evaluate()
+        evaluator = PythonCodeEvaluator(self.test_case_data, self.test, self.language, user_answer, ref_code_path=None, in_dir=None)
+        result = evaluator.evaluate()
         self.assertFalse(result.get("success"))
         self.assertEqual(result.get("error"), "AssertionError  in: assert add(3, 2) == 5")
 
     def test_infinite_loop(self):
-        user_answer = "def add(a, b):\n\twhile True:\n\t\tpass"""
-        test_case_data = [{"func_name": "add", 
-                                 "expected_answer": "5", 
-                                 "test_id": u'null', 
-                                 "pos_args": ["3", "2"], 
-                                 "kw_args": {}
+        user_answer = "def add(a, b):\n\twhile True:\n\t\tpass"
+        timeout_msg = ("Code took more than {0} seconds to run. "
+                       "You probably have an infinite loop in your code.").format(SERVER_TIMEOUT)
+        evaluator = PythonCodeEvaluator(self.test_case_data, self.test, self.language, user_answer, ref_code_path=None, in_dir=None)
+        result = evaluator.evaluate()
+        self.assertFalse(result.get("success"))
+        self.assertEquals(result.get("error"), timeout_msg)
+
+    def test_syntax_error(self):
+        user_answer = "def add(a, b);\n\treturn a+b"
+        syntax_error_msg = ('Traceback (most recent call last):\n  File '
```

Comment:
Again, I already told you in person that this seems fragile.  Isn't there a better way to do this?